src/roompanel/RoomWidget.cpp

```diff
@@ -0,0 +1,252 @@
+        }
+        case Qt::ForegroundRole:
+            if (isFightingYOU(row, column)) {
+                // highlight "YOU" in fighting column
+                return QColor(Qt::red);
+            }
+            return QColor(Qt::black);
+        default:
+            break;
+        }
+    }
+    return empty;
+}
+
+SharedRoomMob RoomModel::getMob(const int row) const
+{
+    if (row >= 0 && size_t(row) < m_mobVector.size()) {
+        return m_mobVector.at(size_t(row));
+    }
+    return nullptr;
+}
+
+const QVariant &RoomModel::getMobField(const int row, const int column) const
+{
+    SharedRoomMob mob = getMob(row);
+    if (!mob) {
+        return empty;
+    }
+    RoomMob::Field i;
+    switch (static_cast<ColumnTypeEnum>(column)) {
+    case ColumnTypeEnum::NAME:
+        i = RoomMob::Field::NAME;
+        break;
+    case ColumnTypeEnum::LABEL:
+        i = RoomMob::Field::LABELS;
+        break;
+    case ColumnTypeEnum::POSITION:
+        i = RoomMob::Field::POSITION;
+        break;
+    case ColumnTypeEnum::EFFECTS:
+        i = RoomMob::Field::FLAGS;
+        break;
+    case ColumnTypeEnum::WEAPON:
+        i = RoomMob::Field::WEAPON;
+        break;
+    case ColumnTypeEnum::FIGHTING:
+        i = RoomMob::Field::FIGHTING;
+        break;
+    case ColumnTypeEnum::MOUNT:
+        i = RoomMob::Field::MOUNT;
+        break;
+    default:
+        return empty;
+    }
+    QVariant &variant = mob->field(i);
+    if (static_cast<QMetaType::Type>(variant.type()) != QMetaType::UInt) {
+        return variant;
+    }
+
+    // field contains the ID of another mob: try to resolve it
+    const auto iter = m_mobsById.find(variant.toUInt());
+    if (iter != m_mobsById.end()) {
+        const SharedRoomMob mob2 = iter->second;
+        if (mob2) {
+            return mob2->field(RoomMob::Field::NAME);
+        }
+    }
+    if (false) {
+        qDebug() << "mob id" << variant << "not found";
+    }
+    // failed to resolve ID of another mob:
+    // clear the ID before MUME reuses it
+    variant = empty;
+    return empty;
+}
+
+QColor RoomModel::getMobColor(const int row) const
+{
+    SharedRoomMob mob = getMob(row);
+    if (mob) {
+        return mob->color();
+    }
+    return QColor{};
+}
+
+bool RoomModel::isFightingYOU(const int row, const int column) const
+{
+    if (static_cast<ColumnTypeEnum>(column) == ColumnTypeEnum::FIGHTING //
+        && row >= 0 && size_t(row) < m_mobVector.size()) {
+        //
+        const SharedRoomMob mob = m_mobVector.at(size_t(row));
+        return mob && mob->field(RoomMob::Field::FIGHTING) == RoomMob::YOU;
+    }
+    return false;
+}
+
+void RoomModel::update()
+{
+    beginResetModel();
+    m_room.listMobs(m_mobsById, m_mobVector);
+    endResetModel();
+}
+
+// ------------------------------- RoomWidget ----------------------------------
+RoomWidget::RoomWidget(RoomManager *rm, QWidget *const parent)
+    : QWidget{parent}
+    , m_model{nullptr}
```

Comment:
Don't initialize here then since we're doing it below

Reply:
Done.